codecs/avif_enc/avif_enc.cpp

```diff
@@ -0,0 +1,88 @@
+#include "avif/avif.h"
+#include <emscripten/bind.h>
+#include <emscripten/val.h>
+
+using namespace emscripten;
+
+struct AvifOptions {
+  // [0 - 63]
+  // 0 = lossless
+  // 63 = worst quality
+  int minQuantizer;
+  int maxQuantizer;
+  // [0 - 6]
+  // Creates 2^n tiles in that dimension
+  int tileRowsLog2;
+  int tileColsLog2;
+  // [0 - 10]
+  // 0 = slowest
+  // 10 = fastest
+  int speed;
+  // 0 = 4:2:0
+  // 1 = 4:2:2
+  // 2 = 4:4:4
+  int subsample;
+};
+
+avifRWData output = AVIF_DATA_EMPTY;
+val encode(std::string buffer, int width, int height, AvifOptions options) {
+  int depth = 8;
+  avifPixelFormat format;
+  switch (options.subsample) {
+  case 0:
+    format = AVIF_PIXEL_FORMAT_YUV420;
```

Comment:
Let's expose the actual underlying enum instead.

Reply:
Yeah, I'll pick this up (and as many other options as I can find) when I do the UI